bin/osg-topology

```diff
@@ -166,48 +166,148 @@ def list_security_contacts(args):
                             if contact_list.tag != 'ContactList':
                                 continue
                             for child in contact_list:
                                 if child.tag == 'ContactType':
-                                    contact_list_type = child.text
+                                    contact_list_type = child.text.lower()
                                 if child.tag == 'Contacts':
                                     for contact in child:
-                                        contact_info = {}
+                                        contact_info = \
+                                            { 'ContactType' : contact_list_type }
                                         for contact_contents in contact:
                                             contact_info[contact_contents.tag] = \
                                                 contact_contents.text
                                         contact_list_info.append(contact_info)
-                            if contact_list_type == 'Security Contact':
-                                security_contacts = contact_list_info
-
-                if name and security_contacts and (not args.name_filter or fnmatch.fnmatch(name, args.name_filter) or args.name_filter in name):
-                    keys = results.setdefault(name, set())
-                    for contact in security_contacts:
-                        key = None
-                        if 'Name' in contact and 'Email' in contact:
-                            key = "%s<%s>" % (contact['Name'], contact['Email'])
-                        elif 'Name' in contact:
-                            key = contact['Name']
-                        if key:
-                            keys.add(key)
-
-    sites = results.keys()
-    sites.sort()
-    for site in sites:
-        keys = list(results[site])
-        keys.sort()
-        print("- Resource: %s" % site)
-        for key in keys:
-             print(u"  * %s" % key)
+
+                if name and contact_list_info:
+                    results[name] = contact_list_info
+
+    print_contacts(args, 'Resource', results)
+
+def list_vo_contacts(args):
+    """
+    List vo contacts for OSG.
+    """
+    base_url = "https://my.opensciencegrid.org/vosummary/xml" \
+                "?summary_attrs_showcontact=on&all_vos=on" \
+                "&active_value=1&active=on&disable=on&disable_value=0"
+
+    with get_auth_session(args) as session:
+        url = mangle_url(base_url, args, session)
+        #print(url)
+        response = session.get(url)
+
+    if response.status_code != requests.codes.ok:
+        print("MyOSG request failed (status %d): %s" % (response.status_code,
+              response.text[:2048]), file=sys.stderr)
+        return 1
+
+    root = ET.fromstring(response.content)
+    if root.tag != 'VOSummary':
+        print("MyOSG returned invalid XML with root tag %s" % root.tag,
+              file=sys.stderr)
+        return 1
```

Comment:
Can we refactor grabbing and verifying the XML into its own function? The first bit is almost exactly the same between the VO and RG functions

Reply:
I was first going to do that but I decided it wasn't worth it because the common piece was so small.